substrate/frame/staking-async/runtimes/rc/src/lib.rs

```diff
@@ -606,19 +606,129 @@ enum RcClientCalls<AccountId> {
 }
 
 pub struct XcmToAssetHub<T: SendXcm, AssetHubId: Get<u32>>(PhantomData<(T, AssetHubId)>);
-impl<T: SendXcm, AssetHubId: Get<u32>> ah_client::SendToAssetHub for XcmToAssetHub<T, AssetHubId> {
-	type AccountId = AccountId;
 
-	fn relay_session_report(session_report: rc_client::SessionReport<Self::AccountId>) {
-		let message = Xcm(vec![
+impl<T: SendXcm, AssetHubId: Get<u32>> XcmToAssetHub<T, AssetHubId> {
+	/// This is a downward message.
+	/// Westend: `51200` (50K)
+	/// Polkadot: `51200` (50K)
+	/// Kusama: `51200` (50K)
+	fn split_until_validated_session_report(
```

Comment:
maybe TODO issue for deduplicating the splitting code between RC and parachain

I don't think we need a splitter trait or anything fancy. Afaict the splitting logic can be a simple utility fn that takes `<T as SendXcm>::validate()` and `session_message_from_report()` as fn params. The rest of the logic looks the same.

Reply:
Same as https://github.com/paritytech/polkadot-sdk/pull/8422/files#diff-a4a0fc985067e6027a30aa3f4c199456c11092f764b7c3313c3f7d26ec762308R331